BraveShared/BraveCertificateUtils.swift

```diff
@@ -0,0 +1,198 @@
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+
+public struct BraveCertificateUtils {
+  /// Formats a hex string
+  /// Example: formatHex("020401") -> "02 04 01"
+  /// Example: formatHex("020401", separator: " - ") -> "02 - 04 - 01"
+  public static func formatHex(_ hexString: String, separator: String = " ") -> String {
+    let n = 2
+    let characters = Array(hexString)
+
+    var result: String = ""
+    stride(from: 0, to: characters.count, by: n).forEach {
+      result += String(characters[$0..<min($0 + n, characters.count)])
+      if $0 + n < characters.count {
+        result += separator
+      }
+    }
+    return result
+  }
+
+  public static func formatDate(_ date: Date) -> String {
+    let dateFormatter = DateFormatter()
+    dateFormatter.dateStyle = .full
+    dateFormatter.timeStyle = .full
+    return dateFormatter.string(from: date)
+  }
+}
+
+public extension BraveCertificateUtils {
+  private enum OIDConversionError: String, Error {
+    case tooLarge
+    case invalidRootArc
+    case invalidBEREncoding
+    case invalidLength
+    case invalidHexDigits
+  }
+
+  /// Converts an ASN1 dot notation OID String to its Binary equivalent
+  static func absolute_oid_to_oid(oid: String) throws -> [UInt8] {
+    var list = [UInt64]()
+    for value in oid.split(separator: ".") {
+      if let result = UInt64(value, radix: 10) {
+        list.append(result)
+      } else {
+        // We don't support larger than 64-bits per arc as it would require big-int.
+        throw OIDConversionError.tooLarge
+      }
+    }
+
+    let encode_octet_as_septet = { (octet: UInt64) -> [UInt8] in
+      var octet = octet
+      var encoded = [UInt8]()
+      var value = UInt64(0x00)
+
+      while octet >= 0x80 {
+        encoded.insert(UInt8((octet & UInt64(0x7F)) | value), at: 0)
+        octet >>= 7
+        value = 0x80
+      }
+      encoded.insert(UInt8(octet | value), at: 0)
+      return encoded
+    }
+
+    if list.count < 2 {
+      throw OIDConversionError.invalidBEREncoding
+    }
+
+    // Invalid encoding for the root arcs 0 and 1.
+    // Invalid encoding the root arc is limited to 0, 1, and 2.
+    if (list[0] > 2) || (list[0] <= 1 && list[1] > 39) {
+      throw OIDConversionError.invalidRootArc
+    }
+
+    var result = encode_octet_as_septet(list[0] * 40 + list[1])
+    for i in 2..<list.count {
+      result.append(contentsOf: encode_octet_as_septet(list[i]))
+    }
+
+    result.insert(UInt8(result.count), at: 0)
+    result.insert(0x06, at: 0)
+    return result
+  }
+
+  /// Converts a Binary OID to its ASN1 dot notation equivalent
+  static func oid_to_absolute_oid(oid: [UInt8]) throws -> String {
+    // Invalid BER encoding
+    if oid.count < 2 {
+      throw OIDConversionError.invalidBEREncoding
+    }
+
+    #if DEBUG
+    // Length octet validation
```

Comment:
Why this validation is for debug builds only?

Reply:
Ah because it's a very strict validation that I use for only making sure it works in a unit test or when debugging it. I don't want this validation done in release.